ns3_adapter/src/ns3_client.cpp

```diff
@@ -113,11 +162,40 @@ void NS3Client::close() {
     work_.reset();
     io_->stop();
     io_thread_->join();
-    udp_listener_->stop();
-    udp_out_socket_.reset();
+    udp_v2x_listener_->stop();
+    udp_time_listener_->stop();
+    udp_out_broadcasting_socket_.reset();
+    udp_out_registration_socket_.reset();
     onDisconnect();
 }
 
+void NS3Client::process_time(const std::shared_ptr<const std::vector<uint8_t>>& data)
+{
+    const std::vector<uint8_t> vec = *data;
+    ROS_DEBUG_STREAM("process_time data received of size: " << vec.size());
+
+    std::string json_string(vec.begin(), vec.end());
+    ROS_DEBUG_STREAM("process_time data: " << json_string);
+
+    // JSON
+    rapidjson::Document obj;
+    std::string timestep_member_name = "timestep";
+    obj.Parse(json_string);
+    if (obj.HasParseError())
+    {
+        throw std::runtime_error(std::string("Message JSON is misformatted. JSON parsing failed! Please check process_time data: " + json_string));
```

Comment:
Do you need the std::string here? I think you should be able to just
```cpp
throw std::runtime_error("Message JSON is misformatted. JSON parsing failed! Please check process_time data: " + json_string);
```

Reply:
You are right, I removed it